viewing_party/party.py

```diff
@@ -1,23 +1,174 @@
+"""
+Dear Ashley (or Instructor),
+
+Can you please take a look at the comments in my wave 3.
+I tried doing something that didn't work out and so had to 
+abandon it all together. I can't figure out what the problem
+is and would appreciate a look.
+
+Also, answering my questions on Wednesday!
+And WOOT WOOT for finishing something I thought would even 
+be able to do halfway!!!!!
+
+Thank you so much,
+Huma Hameed
+(she/her)
+"""
+
+
+
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {}
+    new_movie["title"] = title
+    new_movie["genre"] = genre
+    new_movie["rating"] = rating
+
+    for key, value in new_movie.items():
+        print(new_movie)
+        if value == None:
+            return None
+    return new_movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"] = []
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    # for movies["title"] in user_data["watched"] == title:
+    for movies in user_data["watchlist"]:
+        if movies["title"] == title:
+            print(title)
+            user_data["watched"].append(movies)
+            user_data["watchlist"].remove(movies)
+    return user_data
+        
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    sum = 0
+    length = len(user_data["watched"])
+    if length==0:
+        return 0
+    for movie in user_data["watched"]:
+        sum += movie["rating"]
+    avg = sum/ length
+    return avg            
+
+def get_most_watched_genre(user_data):
+    genre_count = {}
+    # if user_data["watched"]==[]:
+    # length = len(user_data["watched"])
+    # if length==0:
+    #     return None
+    if not user_data["watched"]:
+        return None 
+    for movie in user_data["watched"]:
+        genre = movie["genre"] 
+        if genre not in genre_count:
+            genre_count[genre] = 1
+        else:
+            genre_count[genre] += 1
+    sorted_genre = sorted(genre_count.items(), key = lambda kv:kv[1])

```

Comment:
Nice! Another way you might see line 80 written is `most_freq_genre = max(genre_count, key=genre_count.get)`

If you used this second approach, then what happens is the function .get() would be executed on for each key-value pair in genre_count -- get() returns the value and then max returns the highest integer.